### PR TITLE
Fix for typescript compile error

### PR DESCRIPTION
**Pull request by thinkh**
See related issue phovea/phovea_d3#10 and PR https://github.com/phovea/phovea_core/pull/42

### CODE REVIEW

File: src/layout_d3util.ts [hunk 1]
```diff
@@ -58,7 +58,7 @@ class HTMLLayoutElem extends ALayoutElem implements ILayoutElem {
 
   setBounds(x:number, y:number, w:number, h:number) {
     const unit = this.layoutOption('unit', 'px'),
-      doAnimate = this.layoutOption('animate', false) === true;
+      doAnimate = (this.layoutOption<boolean>('animate', false) === true);
```
Comment (mccalluc) [inline src/layout_d3util.ts]: Do we even need the `=== true`? If the value might have only been true-ish, I could understand, but this seems redundant now?

Comment (thinkh) [reply src/layout_d3util.ts]: Yes, true. ;-) You can remove it.

File: src/link_representation.ts [hunk 2]
```diff
@@ -59,9 +59,9 @@ export function createGroupRep(context: IBandContext, a: IVisWrapper, aa: Rect,
         loc : locs[i] ? locs[i].aabb() : null
       }; };
     }
-    var groupa = data[0].groupa.map(more(toArray(data[1])));
-    var groupb = data[0].groupb.map(more(toArray(data[2])));
-    var r = [];
+    const groupa = (<any>data[0]).groupa.map(more(toArray(data[1])));
+    const groupb = (<any>data[0]).groupb.map(more(toArray(data[2])));
```
Comment (mccalluc) [inline src/link_representation.ts:63]: The `<any>` doesn't seem to provide any new information... Is it necessary here? (I think there's a compile option that requires a type declaration on everything, but I think we have that turned off.)

Comment (thinkh) [reply src/link_representation.ts:63]: Try to checkout all tslint fix branches and your branch. At the end it worked in my local environment.

**Comment by mccalluc** [comment]
We're still failing:
```
node_modules/phovea_core/src/data.ts(164,10): error TS2322: Type 'Promise<IDataType[]>' is not assignable to type 'Promise<IDataType>'.
  Type 'IDataType[]' is not assignable to type 'IDataType'.
    Property 'desc' is missing in type 'IDataType[]'.
```
I would like Travis to pass before merging.

... but beyond that, what changed that the build no longer works? I don't think we've been actively developing it, so my guess is that an unpinned dependency changed, and we're only noticing it now. If this is a correct understanding, a better fix would be to pin that dependency without making other changes to the source (though they could come in later).

**Comment by thinkh** [comment]
@mccalluc Yes. Travis in phovea_d3 will fail because we haven't merged PR https://github.com/phovea/phovea_core/pull/42, which fixes the problem.

**Pull request merged**